File: lab3/src/main/java/lab3/myquest/locations/Corner.java
package lab3.myquest.locations;

import lab3.functionality.Action;
import lab3.functionality.Item;
import lab3.functionality.Location;
import lab3.functionality.Player;
import lab3.myquest.items.BreakableItem;

public class Corner extends Location {
    public Corner(String name) {
        super(name, "Угол непонятной комнаты, которым заканчивается длинная стена.");
        addItem(new BreakableItem("Люстра", "Красивая хрустальная люстра"));
        addItem(new BreakableItem("Лампа", "Яркая лампа"));
        var bookShell = new Item("Книжная полка", "Полка с множеством книг.");
        bookShell.addAction(new Action("Осмотреть", p -> p.getName() + " врезался головой в книжную полку, отчего книги разлетелись в разные стороны.", p -> bookShell.setDescription("Теперь уже пустая полка"), true));
        addItem(bookShell);
    }

    @Override
    public String enter(Player player) {
        player.damage();
        return player.getName() + " ударился спиной об стену, отскочил от нее, словно мячик, и полетел к противоположной стене";
    }
}

File: lab4/src/main/kotlin/lab4/myQuest/locations/Corner.kt
package lab4.myQuest.locations

import lab4.functionality.ActionFactory
import lab4.functionality.Location
import lab4.functionality.Player
import lab4.myQuest.items.BookShell
import lab4.myQuest.items.BreakableItem

class Corner : Location("Угол комнаты", { "Угол непонятной комнаты, которым заканчивается длинная стена." }) {
    init {
        items.add(BreakableItem("Люстра") { "Хрустальная люстра" })
        items.add(BreakableItem("Лампа") { "Яркая лампа" })
        items.add(BookShell())

    }

    override fun enter(player: Player): String {
        player.actions.add(
            ActionFactory().changeLocation(
                "Отшвырнуть стул",
                { "Увидев, что от стула никакой пользы нет, ${it.name} отшвырнул его от себя. В результате стул полетел вниз и, ударившись о пол, подскочил кверху, словно резиновый, сам же ${it.name} отлетел к потолку и, отскочив от него, полетел вниз. По пути он столкнулся с летящим навстречу стулом и получил удар спинкой стула прямо по переносице. Удар был настолько силен, что Знайка ошалел от боли и на некоторое время перестал трепыхаться в воздухе. " },
                Unconscious(), true
            )
        )
        return ""
    }

}